src/private.rs

```diff
@@ -72,100 +61,118 @@ impl RedactFlags {
                 if char.is_alphanumeric() {
                     if prefix_gas > 0 {
                         prefix_gas -= 1;
-                        redacted.push(char);
+                        fmt.write_char(char)?;
```

Comment:
(moved from) https://github.com/primait/veil/pull/61#discussion_r1049681769
I think it's fine actually. It's up for the writer to implement buffering and we shouldn't concern ourselves with it. If someone needs buffering and it's not implemented by the writer they should wrap it in BufWriter instead

Reply:
Agreed, I think it's idiomatic too, there's examples in std of this exact kind of usage:

https://github.com/rust-lang/rust/blob/397b66e77b279de5006facf87979f9ecff5c7f87/library/core/src/str/lossy.rs#L78

https://github.com/rust-lang/rust/blob/397b66e77b279de5006facf87979f9ecff5c7f87/library/std/src/error.rs#L564